src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -3,13 +3,150 @@ Tests for external textures from HTMLVideoElement (and other video-type sources?
 
 - videos with various encodings, color spaces, metadata
 
-TODO: get test videos from WebGL CTS
-TODO(kainino0x): (and set up the build to deal with data files)
 TODO: consider whether external_texture and copyToTexture video tests should be in the same file
-TODO: plan
+
+Test Coverage:
+  - Tests that we can import an HTMLVideoElement into a GPUExternalTexture and sample from it for:
+      - All supported video formats {vp8, vp9, ogg, mp4}
+      TODO: Multiplanar scenarios
 `;
 
 import { makeTestGroup } from '../../../common/framework/test_group.js';
 import { GPUTest } from '../../gpu_test.js';
 
+const kHeight = 16;
+const kWidth = 16;
+const kFormat = 'rgba8unorm';
+const kVideoSources: string[] = [
+  'red-green.webmvp8.webm',
+  'red-green.bt601.vp9.webm',
+  'red-green.mp4',
+  'red-green.theora.ogv',
+];
+
 export const g = makeTestGroup(GPUTest);
+
+g.test('importExternalTexture,sample')
+  .desc(
+    `
+Test that importing different video formats into an external texture and sampling from them works correctly.
+`
+  )
+  .params(u =>
+    u //
+      .combine('videoSource', kVideoSources)
+  )
+  .fn(async t => {
+    const videoUrl = '../out/resources/' + t.params.videoSource;
```

Comment:
I don't think this URL is going to work when we're in WPT. Fortunately, for web_platform, tests we don't have to worry about other runtimes (cmdline (node) and deno runtimes), but we still have both the standalone and wpt runtimes. And the wpt runtime is the one we use on automated testing, so we'll want it to work.

We may need some kind of base URL (pointing to the root of the src/ or out/ or out-wpt/ directory or whatever), passed in from the runtime, so that we know how to grab these resources. Happy to chat more about this, or look into it (but may take me some time to get to it).

Reply:
I see - so it looks like I can acquire `window.location.href` within common/runtime/wpt.ts and then pass it into `run()` - right? Alternatively is there any reason I can't just use `window.location.href` directly from the test?

Reply:
Unfortunately window.location.href by itself won't let us figure out the location of the resources directory. The file paths look like this:

- In standalone: `/standalone/` -> `/out/resources/`
- In the actual WPT build: `/webgpu/cts.html` -> `/webgpu/resources/`
- In Chromium: `/wpt_internal/webgpu/cts.html` -> `/gen/third_party/webgpu-cts/src/`

In order to deal with this for the rest of the generated files, we point to the appropriate build directory from the template:
- https://github.com/gpuweb/cts/blob/main/src/common/templates/cts.html#L32
- https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/webgpu/ctshtml-template.txt;l=24;drc=6b69efe99ef855e2ac953b37a31063d07ecfee9a

So I think we are going to have to add something like `loadWebGPUExpectations` to point to the root of the build directory.

Reply:
Resolved.